cupyx/linalg/solve.py

```diff
@@ -7,6 +7,75 @@
 from cupy.linalg import util
 
 
+def _batched_invh(a):
+    """Compute the inverse of an array of Hermitian matrices.
+
+    This function computes an inverse of a real symmetric or complex hermitian
+    positive-definite matrix using Cholesky factorization. If matrix ``a[i]``
+    is not positive definite, Cholesky factorization fails and it raises
+    an error.
+
+    Args:
+        a (cupy.ndarray): Array of real symmetric or complex hermitian
+            matrices with dimension (..., N, N).
+
+    Returns:
+        cupy.ndarray: The array of inverses of matrices ``a[i]``.
+    """
+    if a.dtype.char == 'f' or a.dtype.char == 'd':
+        dtype = a.dtype.char
+    else:
+        dtype = numpy.promote_types(a.dtype.char, 'f').char
+
+    if dtype == 'f':
+        potrfBatched = cusolver.spotrfBatched
+        potrsBatched = cusolver.spotrsBatched
+    elif dtype == 'd':
+        potrfBatched = cusolver.dpotrfBatched
+        potrsBatched = cusolver.dpotrsBatched
+    elif dtype == 'F':
+        potrfBatched = cusolver.cpotrfBatched
+        potrsBatched = cusolver.cpotrsBatched
+    elif dtype == 'D':
+        potrfBatched = cusolver.zpotrfBatched
+        potrsBatched = cusolver.zpotrsBatched
+    else:
+        msg = ('dtype must be float32, float64, complex64 or complex128'
+               ' (actual: {})'.format(a.dtype))
+        raise ValueError(msg)
+
+    a = a.astype(dtype, order='C', copy=True)
+    ap = cupy.core.core._mat_ptrs(a)
+    n = a.shape[-1]
+    lda = a.strides[-2] // a.dtype.itemsize
+    handle = device.get_cusolver_handle()
+    uplo = cublas.CUBLAS_FILL_MODE_LOWER
+    batch_size = int(numpy.prod(a.shape[:-2]))
+    dev_info = cupy.empty(batch_size, dtype=numpy.int32)
+
+    # Cholesky factorization
+    potrfBatched(handle, uplo, n, ap.data.ptr, lda, dev_info.data.ptr,
+                 batch_size)
+    cupy.linalg.util._check_cusolver_dev_info_if_synchronization_allowed(
+        potrfBatched, dev_info)
+
+    identity_matrix = cupy.eye(n, dtype=dtype)
+    b = cupy.tile(identity_matrix, (*a.shape[:-2], 1, 1))
```

Comment:
```suggestion
    b = cupy.empty(a.shape, dtype)
    b[...] = identity_matrix
```

Reply:
Thank you for the suggestion! It indeed looks cleaner than using `cupy.tile`.